File: bootcamps/python_ai_backend_dev/python_db/create.py
from common import get_conn, get_cursor
from sqlite3 import Cursor


def create_table(cursor: Cursor):
    try:
        cursor.execute(
            """
CREATE TABLE IF NOT EXISTS
    clientes (
        id INTEGER PRIMARY KEY AUTOINCREMENT, name VARCHAR(100),
        email VARCHAR(150)
        );
"""
        )
    except Exception as e:
        print(e)


def main() -> None:
    conn = get_conn()
    cur = get_cursor(conn)
    create_table(cur)
    conn.commit()
    conn.close()


if __name__ == "__main__":
    main()

File: golang/dev_book/api/src/db/create.go
package db

import (
	"log"
)

const userTable = `
CREATE TABLE IF NOT EXISTS users(
	id INT AUTO_INCREMENT NOT NULL,
	name VARCHAR(255) NOT NULL,
	user_name VARCHAR(255) NOT NULL UNIQUE,
	email VARCHAR(255) NOT NULL UNIQUE,
	password VARCHAR(255) NOT NULL,
	created_at DATE default current_timestamp(),
	PRIMARY KEY (id)
);
`

const followerTable = `
CREATE TABLE IF NOT EXISTS followers(
	users_id INT NOT NULL,
	FOREIGN KEY (users_id)
	REFERENCES users(id)
	ON DELETE CASCADE,
	follower_id INT NOT NULL,
	FOREIGN KEY (follower_id)
	REFERENCES users(id)
	ON DELETE CASCADE,
	PRIMARY KEY (users_id, follower_id)
);
`

const postsTable = `
CREATE TABLE IF NOT EXISTS posts(
	id INT AUTOINCREMENT NOT NULL,
	title VARCHAR(50) NOT NULL,
	content TEXT NOT NULL,
	author_id INT NOT NULL
	FOREIGN KEY (author_id)
	REFERENCES users(id)
	ON DELETE CASCADE,
	likes INT DEFAULT 0,
	created_at DATE default current_timestamp(),
	PRIMARY KEY (id)
);
`

func CreateTables() {
	conn, err := GetConnection()
	if err != nil {
		log.Fatal(err)
	}
	defer conn.Close()
	tables := []string{
		userTable, followerTable, postsTable,
	}
	for _, table := range tables {
		if _, err := conn.Query(table); err != nil {
			log.Fatal(err)
		}
	}
}
